src/server/rpc/rpc-manager.js

```diff
@@ -14,9 +14,55 @@ var fs = require('fs'),
     SocketManager = require('../socket-manager'),
     utils = require('../server-utils'),
     JsonToSnapList = require('./procedures/utils').jsonToSnapList ,
-    jsdocExtractor = require('./jsdoc-extractor.js'),
+    Docs = require('./jsdoc-extractor.js').Docs,
     RESERVED_FN_NAMES = require('../../common/constants').RPC.RESERVED_FN_NAMES;
 
+// TODO move this fn and maybe Docs class?
+// in: arg obj and input value
+// out: [parsedInput, msg]
+function parseArgValue(arg, input) {
+    let errorMsg;
+    const genericErr = `${arg.name} = ${input} is not a valid ${arg.type}. `;
+    if (input === '') {
+        if (!arg.isOptional) {
+            errorMsg = `${arg.name} is required.`;
+        }
+        input = undefined;
+        return [input, errorMsg];
+    }
+    // TODO add check for optional
```

Comment:
Don't you already have a check for optional arguments?

Reply:
yep, it's leftover